File: labs/01-first-llm-call/go/main.go
package main

import (
	"fmt"
	"log"

	"github.com/ZYZ-Labs/agent-labs/shared/config"
)

func main() {
	client := openaiclient.NewClient()
	resp, err := client.ChatCompletion(openaiclient.ChatCompletionRequest{
		Messages: []openaiclient.Message{
			{Role: "system", Content: "You are a concise assistant."},
			{Role: "user", Content: "Explain what an AI agent is in one sentence."},
		},
		MaxTokens: intPtr(80),
	})
	if err != nil {
		log.Fatalf("Request failed: %v", err)
	}

	msg := client.ExtractMessage(resp)
	fmt.Println("Assistant:", msg.Content)
	fmt.Println("Finish reason:", resp.Choices[0].FinishReason)
	fmt.Printf("Usage: %+v\n", resp.Usage)
}

func intPtr(v int) *int { return &v }

File: labs/01-first-llm-call/java/Main.java
import com.agentlabs.OpenAiClient;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;

public class Main {
    public static void main(String[] args) throws Exception {
        OpenAiClient client = new OpenAiClient();
        ObjectMapper mapper = new ObjectMapper();

        ObjectNode payload = client.buildMessages("Explain what an AI agent is in one sentence.");
        payload.put("max_tokens", 80);

        JsonNode response = client.chatCompletion(payload);
        JsonNode message = client.extractMessage(response);

        System.out.println("Assistant: " + message.get("content").asText());
        System.out.println("Finish reason: " + response.get("choices").get(0).get("finish_reason").asText());
        System.out.println("Usage: " + mapper.writeValueAsString(response.get("usage")));
    }
}
